azure-pipelines.yml

```diff
@@ -94,6 +94,24 @@ jobs:
       fi
     displayName: Lint Verilog source files with Verilator
 
+  - bash: |
+      # Build CSR testbench
+      fusesoc --cores-root=. run --setup --build --target=sim lowrisc:ibex:tb_cs_registers
+      if [ $? != 0 ]; then
+        echo -n "##vso[task.logissue type=error]"
+        echo "CS registers TB failed to build. Run 'fusesoc --cores-root=. run --setup --build --target=sim lowrisc:ibex:tb_cs_registers' to check and fix all errors."
```

Comment:
I know the lint task has a simple output, but I think this is probably an anti-pattern. There's risk that the printed command and what is actually executed drifts.

Using `set -x` would be an alternative. It seems there is a [negative interaction](https://docs.microsoft.com/en-us/azure/devops/pipelines/troubleshooting?view=azure-devops) with setting variables via `##vso`, but we're not doing that in these definitions currently. Or we put the command in a string and then eval it or do `bash -c`, but this is getting increasingly ugly...

Reply:
We used this construct as workaround in OT mainly because people didn't have access to the actual build logs. So I'd simply remove the echo parts.

The second problem is that Verilator/fusesoc run doesn't return a non-zero exit code if the run fails. The nicest way would probably be an extension to tb_cs_registers to get this information into the return code. If that's too annoying  I'd go for a wrapper script which does essentially the tee and grep-thing.